### ISSUE TITLE
Build fails if PATH_MAX is not defined

### ISSUE THREAD

**Issue by jannic** [mongrel2/mongrel2#305]
PATH_MAX is not required by POSIX, and it's not defined on GNU Hurd:
https://www.gnu.org/software/hurd/community/gsoc/project_ideas/maxpath.html

mongrel2 uses PATH_MAX in a few places. Building it fails on GNU Hurd.
(https://buildd.debian.org/status/fetch.php?pkg=mongrel2&arch=hurd-i386&ver=1.11.0-1&stamp=1458009555)

In Unixy_getcwd(), get_current_dir_name() could be used, on glibc targets. Or just use a sufficiently large value instead of PATH_MAX - if the cwd is extraordinarily long and exceeds the chosen value, getcwd will return NULL, so nothing (too) bad happens.

normalize_path() in src/dir.c is more tricky: for url_decode, one could allocate strlen(bdata(target)) bytes, as url_decode never increases the string length, but for realpath() that's not true. And, according to the realpath() man page (http://man7.org/linux/man-pages/man3/realpath.3.html) there seems to be no completely portable alternative.
Perhaps the way to go is using PATH_MAX if it's defined, and using the realpath(..., NULL) extension if it's not?

So, if PATH_MAX is not defined, one could use something like:

```c
static inline int normalize_path(bstring target)
{
    char *path_buf = malloc(blength(target));
    check_mem(path_buf);

    url_decode((const char *)(bdata(target)), path_buf);
    bassigncstr(target, path_buf);
    free(path_buf);

    char *normalized = realpath((const char *)(bdata(target)), NULL);
    check_debug(normalized, "Failed to normalize path: %s", bdata(target));

    bassigncstr(target, normalized);
    free(normalized);

    return 0;

error:
    return -1;
}
```

(That's not tested, therefore no pull request...)

**Comment by jasom** [comment]
Unixy_getcwd() is actually no longer used, so I'll just delete it.

Using NULL in realpath() if PATH_MAX isn't defined is the right solution, and I will make that change.

We actually originally used NULL, as that's worked on glibc/Linux for a long time, but OS X does not support that, so we switched to PATH_MAX.

**Comment by jasom** [comment]
Fixed by #306
